daemon/stats.go

```diff
@@ -43,6 +44,15 @@ func (daemon *Daemon) ContainerStats(ctx context.Context, prefixOrName string, c
 		})
 	}
 
+	// Get container stats directly if OneShot is set
+	if config.OneShot {
+		stats, err := daemon.GetContainerStats(ctr)
```

Comment:
@corhere @cpuguy83 I saw an earlier mention of using singleflight; https://github.com/moby/moby/pull/46448#issuecomment-1724104569

> For 1, I think we can use a singleflight per container.

Would this be the path for which to consider a singleflight, or would that be a premature optimisation?

Reply:
I rebutted against using singleflight in https://github.com/moby/moby/pull/46448#issuecomment-1731942882

Reply:
oh! I saw the part about caching, but missed this part;

> ... and not coalesced with other requests.

all good 👍